### PR TITLE
Save out multiple files per task and tweak Stokes I RFI task

### PR DESCRIPTION
**Pull request by ljgray**
Three updates:
1. Add functionality for tasks to save multiple outputs
    - The ability for a task to pass along more than one output was added in radiocosmology/caput#258. This update also allows tasks to *save* multiple outputs
2. Remove the second MAD over bright transits and daytime data from the Stokes I RFI mask. This was not behaving well near the edges of heavily flagged regions, and sensitivity-based flagging should be able to catch any RFI that gets through. Also, save out the mean power metric which is used in the sum threshold flagging step so that we can look at it.
3. Update the implementation of `sumthreshold`. This update is mostly to simplify the code, but it also offers a 10-25% speed improvement. NOTE: `ndimage.convolve1d` takes the negative of `origin` and calls `correlate1d` under the hood. By using `correlate1d` directly, we skip that step, which was a bit confusing. 
    - I don't think there's much that can be done to further speed this up other than parallelizing the correlations in cython.

### CODE REVIEW

**Review by sjforeman — APPROVED** [review]
The SumThreshold reference link is broken, but this one works: https://andreoffringa.org/pdfs/SumThreshold-technical-report.pdf

**Review by ssiegelx — APPROVED** [review]
Having the capability to output and save multiple containers will be incredibly useful!

File: draco/analysis/flagging.py
Comment (ljgray) [reply draco/analysis/flagging.py:1216]: I did some testing with the raw difference originally, but I actually found that it performed worse in that case. However, that was before the sumthrehsold alignment fix, and I did not apply the SIR operator over frequencies, so maybe I'll revisit it.

File: draco/analysis/flagging.py
Comment (ljgray) [reply draco/analysis/flagging.py:1209]: Yeah, this is one of the issues with this approach that I haven't been able to resolve. Even after applying the low-pass filter, there is some residual power from the galaxy and point sources appearing in sidelobes. I experimented with various window sizes, and I found that this residual power changes fast enough in frequency that, with larger windows, it would end up starting to mask some areas that were free of RFI. 

In the testing that I did, the only significant issue was the Russian satellite, since that's a larger blob of RFI. I had previously included the extra convolution with a gaussian kernel to try to fill in unmasked data around that, but I decided to remove that because the new sensitivity and chi-squared rfi masks are able to handle that sort of RFI better than this method.

**Comment by ljgray** [comment]
@ssiegelx I'm going to wait until #274 is merged to merge this one. There are going to be some merge conflicts that I think will be easier to resolve by rebasing this PR onto yours instead of vice verse

File: draco/analysis/flagging.py
Comment (ljgray) [reply draco/analysis/flagging.py]: I investigated this. It seems now like the behaviour of flagging on the raw difference with `max_m=16` is effectively the same as flagging the absolute difference with `max_m=32`, and flagging on the raw difference has slightly better behaviour at the edges of fully masked bands. So, I've switched to using just the raw difference

**Comment by ssiegelx** [comment]
Sounds good, I'll merge that now.

**Pull request merged**
